shared-modules-auto-pr/entrypoint.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+
+import os
+import urllib
+import contextlib
+import tempfile
+import shutil
+import pathlib
+import subprocess
+import logging
+import re
+
+import yaml
+from github import Github
+import pygit2
+from gi import require_version
+require_version('Json', '1.0')
+from gi.repository import Json
+
+SHARED_MODULES = 'shared-modules'
```

Comment:
There are applications not using `shared-module` as the submodule name. It would be better to check for submodule URL.

Reply:
Good catch, fixed

Reply:
I think it's not solved yet. `find_shared_submodules` should be returning the name of directory at which `shared-modules` is added (or `None`) and used as `SHARED_MODULES` value. Otherwise all `repo.lookup_submodule` and git operations will trying to update it will fail.